interfaces/cython/cantera/cti2yaml.py

```diff
@@ -0,0 +1,1648 @@
+                 transport='None', initial_state=None, options=[]):
+
+        ideal_interface.__init__(self, name, elements, species, note, reactions,
+            site_density, phases, kinetics, transport, initial_state, options)
+        self.thermo_model = 'edge'
+
+
+# Falloff parameterizations
+
+class Troe(object):
+    """The Troe falloff function."""
+    def __init__(self, A=0.0, T3=0.0, T1=0.0, T2=None):
+        """
+        Parameters: *A*, *T3*, *T1*, *T2*. These must be entered as pure
+        numbers with no attached dimensions.
+        """
+        self.A = A
+        self.T3 = T3
+        self.T1 = T1
+        self.T2 = T2
+
+    def get_yaml(self, out):
+        troe = FlowMap([('A', self.A), ('T3', self.T3), ('T1', self.T1)])
+        if self.T2 is not None:
+            troe['T2'] = self.T2
+        out['Troe'] = troe
+
+
+class SRI(object):
+    """ The SRI falloff function."""
+    def __init__(self, A=0.0, B=0.0, C=0.0, D=None, E=None):
+        """
+        Parameters: *A*, *B*, *C*, *D*, *E*. These must be entered as
+        pure numbers without attached dimensions.
+        """
+        self.A = A
+        self.B = B
+        self.C = C
+        self.D = D
+        self.E = E
+
+    def get_yaml(self, out):
+        sri = FlowMap([('A', self.A), ('B', self.B), ('C', self.C)])
+        if self.D is not None and self.E is not None:
+            sri['D'] = self.D
+            sri['E'] = self.E
+        out['SRI'] = sri
+
+
+class Lindemann(object):
+    """The Lindemann falloff function."""
+    def get_yaml(self, out):
+        pass
+
+
+def convert(filename=None, outName=None, text=None):
+    # Reset global state, in case cti2yaml is being used as a module and convert
+    # is being called multiple times.
+    units('m', 'kmol', 'kg', 's', 'J/kmol', 'J', 'Pa')
+    standard_pressure(OneAtm)
+    global _motz_wise
+    _motz_wise = None
+    _elements.clear()
+    _species.clear()
+    _speciesnames.clear()
+    _phases.clear()
+    _reactions.clear()
+    _reactions['reactions'] = []
+
+    if filename is not None:
+        filename = os.path.expanduser(filename)
+        base = os.path.basename(filename)
+        root, _ = os.path.splitext(base)
+        dataset(root)
+    if outName is None and _name != 'noname':
+        outName = _name + '.yaml'
+
+    open_kw = {'encoding': 'latin-1'} if sys.version_info.major == 3 else {}
+    open_mode = 'r' if sys.version_info.major == 3 else 'rU'
+    try:
+        if filename is not None:
+            with open(filename, open_mode, **open_kw) as f:
+                code = compile(f.read(), filename, 'exec')
+        else:
+            code = compile(text, '<string>', 'exec')
+        exec(code)
+    except SyntaxError as err:
+        # Show more context than the default SyntaxError message
+        # to help see problems in multi-line statements
+        if filename:
+            text = open(filename, open_mode).readlines()
```

Comment:
Ideally, should use a context manager here, even for a one liner. That said, since this only needs to work for Python 2 for the build script, and the build script should never hit this code because we know we're giving it good files, you could use `pathlib.Path(filename).read_text().split('\n')` if you imported `pathlib` right here instead of at the top of the file.

Reply:
Or, since the program exits at the end of this block, with `sys.exit(3)`,  maybe it doesn't matter.